enterprise/backup/src/main/java/org/neo4j/backup/BackupService.java

```diff
@@ -257,6 +260,18 @@ BackupOutcome doIncrementalBackupOrFallbackToFull( String sourceHostNameOrIp, in
                         "but that failed as well: '" + fullBackupFailure.getMessage() + "'.", fullBackupFailure );
             }
         }
+        catch ( RuntimeException e )
+        {
+            Throwable cause = e.getCause() == null ? e : rootCause( e );
```

Comment:
That's the wrong `null` case, isn't it? You're checking `e.getCause()`, which it copes with being `null`. The case you point out would be equivalent to `e` being `null`.